### PR TITLE
refactor!: Vaex DType wraps NumPy and Arrow DType in uniform way

### PR DESCRIPTION
**Pull request by maartenbreddels**
See the docstring of datatype.py for usage.
We should decide if we want this or not, and if so, we need to put this also in the docs:

TODO:
 * Add `py.test --doctest-modules packages/vaex-core/vaex/datatype.py` to CI
 * add `packages/vaex-core/vaex/datatype_test.py` to CI
 * Review naming

### CODE REVIEW

**Comment by kmcentush** [comment]
+1, as I believe this would help to address the issues I noticed in https://github.com/vaexio/vaex/issues/1000

**Comment by maartenbreddels** [comment]
Thanks for the feedback @kmcentush , glad you like it.

File: packages/vaex-core/vaex/datatype.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        >>> DataType(pa.large_string()).is_string
+        True
+        >>> DataType(pa.large_string()) == 'string'
+        True
+        '''
+        return vaex.array_types.is_string_type(self.internal)
+
+    # @property
+    # def is_str(self):
+    #     return self.is_string
```
Comment (JovanVeljanoski) [inline packages/vaex-core/vaex/datatype.py:250]: Can this be removed?

**Review by JovanVeljanoski — APPROVED** [review]
Looks great! 

Did some light testing, runs as expected 👍 
Loving it <3

**Pull request merged**